python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -82,6 +82,7 @@ def size(self):
             )
             .groupby(self.grouping)
             .agg("size")
+            .sort_index()
```

Comment:
Why do we need to do a `sort_index` here?

Reply:
This is necessary for functions like `nth`.